cynic-querygen/src/lib.rs

```diff
@@ -128,36 +145,43 @@ pub fn document_to_fragment_structs(
     }
     writeln!(output, "}}\n").unwrap();
 
-    writeln!(output, "#[cynic::query_module(").unwrap();
-    writeln!(output, "    schema_path = r#\"{}\"#,", options.schema_path).unwrap();
-    writeln!(output, "    query_module = \"{}\",", options.query_module).unwrap();
-    writeln!(output, ")]\nmod types {{").unwrap();
-
-    // Output any custom scalars we need.
-    // Note that currently our query_dsl needs _all_ scalars in a schema
-    // so we're parsing this out from schema.definitons rather than output.scalars
-    for def in &schema.definitions {
-        if let schema::Definition::TypeDefinition(schema::TypeDefinition::Scalar(scalar)) = def {
-            writeln!(output, "    #[derive(cynic::Scalar, Debug, Clone)]").unwrap();
-            writeln!(
-                output,
-                "    pub struct {}(pub String);\n",
-                scalar.name.to_pascal_case()
-            )
-            .unwrap();
+    if scalar_count > 0 {
+        writeln!(output, "#[cynic::query_module(").unwrap();
+        writeln!(output, "    schema_path = r#\"{}\"#,", options.schema_path).unwrap();
+        writeln!(output, "    query_module = \"{}\",", options.query_module).unwrap();
+        writeln!(output, ")]\nmod types {{").unwrap();
+
+        // Output any custom scalars we need.
+        // Note that currently our query_dsl needs _all_ scalars in a schema
+        // so we're parsing this out from schema.definitons rather than output.scalars
+        for def in &schema.definitions {
+            if let schema::Definition::TypeDefinition(schema::TypeDefinition::Scalar(scalar)) = def
+            {
+                writeln!(output, "    #[derive(cynic::Scalar, Debug, Clone)]").unwrap();
+                writeln!(
+                    output,
+                    "    pub struct {}(pub String);\n",
+                    scalar.name.to_pascal_case()
+                )
+                .unwrap();
+            }
         }
+        writeln!(output, "}}\n").unwrap();
     }
-    writeln!(output, "}}\n").unwrap();
 
     writeln!(output, "mod {}{{", options.query_module).unwrap();
-    writeln!(output, "    use super::types::*;").unwrap();
+
+    if scalar_count > 0 {
+        writeln!(output, "    use super::types::*;").unwrap();
+    }
+
     writeln!(
         output,
         "    cynic::query_dsl!(r#\"{}\"#);",
         options.schema_path
     )
     .unwrap();
-    writeln!(output, "}}\n").unwrap();
+    writeln!(output, "}}\n").unwrap(); // Why have two \n ?
```

Comment:
Good question.  I think I probably just copied this line from the other lines where we close a module, where I explicitly wanted a blank line between modules.

Absolutely not neccessary on this one though, so feel free to remove 👍